google/auth/identity_pool.py

```diff
@@ -0,0 +1,206 @@
+                    )
+            else:
+                self._credential_source_field_name = None
+        else:
+            self._credential_source_file = None
+        if not self._credential_source_file:
+            raise exceptions.GoogleAuthError("Missing credential_source file")
+
+    @_helpers.copy_docstring(external_account.Credentials)
+    def retrieve_subject_token(self, request):
+        return self._get_token_file(
+            self._credential_source_file,
+            self._credential_source_format_type,
+            self._credential_source_field_name,
+        )
+
+    def _get_token_file(
+        self, filename, format_type="text", subject_token_field_name=None
+    ):
+        if not os.path.exists(filename):
+            raise exceptions.RefreshError("File '{}' was not found.".format(filename))
+
+        with io.open(filename, "r", encoding="utf-8") as file_obj:
+            content = file_obj.read()
+
+        if format_type == "text":
+            token = content
+        else:
+            try:
+                # Parse file content as JSON.
+                response_data = json.loads(content)
+                # Get the subject_token.
+                token = response_data[subject_token_field_name]
+            except (KeyError, ValueError):
+                raise exceptions.RefreshError(
+                    "Unable to parse subject_token from JSON file '{}' using key '{}'".format(
+                        filename, subject_token_field_name
+                    )
+                )
+        if not token:
+            raise exceptions.RefreshError(
+                "Missing subject_token in the credential_source file"
+            )
+        return token
+
+    @classmethod
+    def from_info(cls, info, **kwargs):
+        """Creates an Identity Pool Credentials instance from parsed external account info.
+
+        Args:
+            info (Mapping[str, str]): The Identity Pool external account info in Google
+                format.
+            kwargs: Additional arguments to pass to the constructor.
+
+        Returns:
+            google.auth.identity_pool.Credentials: The constructed
+                credentials.
+
+        Raises:
+            google.auth.exceptions.GoogleAuthError: For invalid parameters.
+        """
+        return cls(
+            audience=info.get("audience"),
+            subject_token_type=info.get("subject_token_type"),
+            token_url=info.get("token_url"),
+            service_account_impersonation_url=info.get(
+                "service_account_impersonation_url"
+            ),
+            client_id=info.get("client_id"),
+            client_secret=info.get("client_secret"),
+            credential_source=info.get("credential_source"),
+            quota_project_id=info.get("quota_project_id"),
+            **kwargs
+        )
+
+    @classmethod
+    def from_file(cls, filename, **kwargs):
```

Comment:
Does it make sense to allow this type of credential to be loaded via `google.auth.load_credentials_from_file`? Currently authorized user (gcloud) and service account are supported.

I can see an argument for not adding it as well - these are external credentials and it looks like it might be hard to detect the file type.

https://github.com/googleapis/google-auth-library-python/blob/ecd88d4f0efc5c619ebd3e3fa7e2472f11c63452/google/auth/_default.py#L72

Reply:
That is the eventual plan and the recommended flow for developers.

We definitely want to support this new type of credential as part of ADC but we also want to provide the ability to explicitly initialize this similar to the ability to initialize service account credentials explicitly from a file or dictionary:
https://github.com/googleapis/google-auth-library-python/blob/ee5617ccfb7caa6ba652c9cc07f716af31eb6ac0/google/oauth2/service_account.py#L216

and

https://github.com/googleapis/google-auth-library-python/blob/ee5617ccfb7caa6ba652c9cc07f716af31eb6ac0/google/oauth2/service_account.py#L195